src/server/oauth2/OAuth2Client.ts

```diff
@@ -207,4 +219,118 @@ export class OAuth2Client {
+            const url = new URL(refreshTokenUri);
+
+            const tokensInfo = await this.getAccessTokenForUser(user);
+
+            for (const tokenInfo of tokensInfo) {
+                if (tokenInfo.refreshToken) {
+                    url.searchParams.set('client_id', clientId);
+                    url.searchParams.set('redirect_uri', `${siteUrl}${redirectUri}`);
+                    url.searchParams.set('refresh_token', tokenInfo.refreshToken);
+                    url.searchParams.set('client_secret', clientSecret);
+                    url.searchParams.set('grant_type', IGrantType.RefreshToken);
+
+                    const { content } = await this.app.getAccessors().http.post(url.href);
+
+                    const { access_token, expires_in, refresh_token } = JSON.parse(
+                        content as string,
+                    );
+
+                    if (access_token) {
+                        await this.removeToken({ userId: user.id, persis }),
+                        await this.saveToken({
+                            token: access_token,
+                            expiresAt: expires_in,
+                            refreshToken: refresh_token,
+                            userId: user.id,
+                            persis,
+                        });
+                    }
+                }
+            }
+        } catch (error) {
+            this.app.getLogger().error(error);
+        }
+
+    }
+
+    public async revokeUserAccessToken({ user, persis }: { user: IUser, persis: IPersistence}) {
+        const tokensInfo = await this.getAccessTokenForUser(user);
+
+        const url = new URL(this.config.revokeTokenURI);
+
+        for (const tokenInfo of tokensInfo) {
+            url.searchParams.set('token', tokenInfo.token);
+
+            await this.app.getAccessors().http.post(url.href);
+        }
+
+        this.removeToken({ userId: user.id, persis });
+
+    }
+
+    private async saveToken({
+        token,
+        expiresAt,
+        refreshToken,
+        userId,
+        persis,
+    }: {
+        token: string,
+        expiresAt: number,
+        refreshToken: string,
+        userId: string,
+        persis: IPersistence,
+    }) {
+        await persis.createWithAssociations(
+            {
+                token,
+                expiresAt: expiresAt || '',
+                refreshToken: refreshToken || '',
+            },
+            [
+                new RocketChatAssociationRecord(
+                    RocketChatAssociationModel.USER,
+                    userId,
+                ),
+                new RocketChatAssociationRecord(
+                    RocketChatAssociationModel.MISC,
+                    `${this.config.alias}-oauth-connection`,
+                ),
+            ],
+        );
+    }
+
+    private async removeToken({ userId, persis }: { userId: string, persis: IPersistence} ) {
+        await persis.removeByAssociations([
```

Comment:
We can replace the `await` for a `return` - as this is the last statement of the function, `await`ing it essentially does nothing, as there will be nothing to run afterwards. By using a `return`, we're deferring the `await`ing to the calling method